### PR TITLE
MutableFragment implementation.

### PR DESCRIPTION
**Pull request by luoxiaojian**

### CODE REVIEW

File: grape/graph/vertex.h [hunk 1]
```diff
@@ -28,119 +28,94 @@ namespace internal {
...
  *
  * @tparam VID_T
  * @tparam VDATA_T
  */
 template <typename VID_T, typename VDATA_T>
-class Vertex {
- public:
-  DEV_HOST Vertex() {}
+struct Vertex {
+  Vertex() = default;
```
Comment (septicmk) [inline grape/graph/vertex.h:39]: use `DEV_HOST Vertex() {}` to avoid warnings

File: grape/graph/vertex.h [hunk 2]
```diff
@@ -28,119 +28,94 @@ namespace internal {
...
-  DEV_HOST explicit Vertex(const VID_T& vid) : vid_(vid), vdata_() {}
+  DEV_HOST explicit Vertex(const VID_T& vid) : vid(vid), vdata() {}
   DEV_HOST Vertex(const VID_T& vid, const VDATA_T& vdata)
-      : vid_(vid), vdata_(vdata) {}
-  DEV_HOST Vertex(const Vertex& vert) : vid_(vert.vid_), vdata_(vert.vdata_) {}
+      : vid(vid), vdata(vdata) {}
+  DEV_HOST Vertex(const Vertex& vert) : vid(vert.vid), vdata(vert.vdata) {}
 
-  DEV_HOST ~Vertex() {}
+  ~Vertex() = default;
```
Comment (septicmk) [inline grape/graph/vertex.h:46]: ditto.

File: grape/graph/vertex.h [hunk 3]
```diff
@@ -28,119 +28,94 @@ namespace internal {
...
 /**
  * @brief Partial specialization for Vertex with EmptyType on vertex_data.
  *
  */
 template <typename VID_T>
-class Vertex<VID_T, EmptyType> {
- public:
-  DEV_HOST Vertex() : vid_() {}
+struct Vertex<VID_T, EmptyType> {
+  Vertex() = default;
```
Comment (septicmk) [inline grape/graph/vertex.h:67]: ditto.

File: grape/graph/vertex.h [hunk 4]
```diff
@@ -28,119 +28,94 @@ namespace internal {
...
 
-  DEV_HOST explicit Vertex(const VID_T& vid) : vid_(vid) {}
-  DEV_HOST Vertex(const VID_T& vid, const EmptyType&) : vid_(vid) {}
-  DEV_HOST Vertex(const Vertex& vert) : vid_(vert.vid_) {}
+  DEV_HOST explicit Vertex(const VID_T& vid) : vid(vid) {}
+  DEV_HOST Vertex(const VID_T& vid, const EmptyType&) : vid(vid) {}
+  DEV_HOST Vertex(const Vertex& vert) : vid(vert.vid) {}
 
-  DEV_HOST ~Vertex() {}
+  ~Vertex() = default;
```
Comment (septicmk) [inline grape/graph/vertex.h:73]: ditto

File: grape/graph/edge.h [hunk 5]
```diff
@@ -33,75 +33,37 @@ class OutArchive;
  * @tparam EDATA_T
  */
 template <typename VID_T, typename EDATA_T>
-class Edge {
- public:
-  DEV_HOST Edge() : src_(), dst_(), edata_() {}
-  DEV_HOST ~Edge() {}
+struct Edge {
+  Edge() : src(), dst(), edata() {}
```
Comment (septicmk) [inline grape/graph/edge.h:37]: use `DEV_HOST Edge() : src(), dst(), edata() {}` to avoid warnings.

File: grape/graph/edge.h [hunk 6]
```diff
@@ -33,75 +33,37 @@ class OutArchive;
...
+  ~Edge() {}
```
Comment (septicmk) [inline grape/graph/edge.h:38]: ditto

**Review by septicmk — APPROVED** [review]

**Review by sighingnow — APPROVED** [review]

**Pull request merged**